### PR TITLE
Implement e2e tests and frontend code

### PR DESCRIPTION
**Pull request by Zaubeerer**

### CODE REVIEW

**Review by bbelderbos — APPROVED** [review]
Thanks, it's common to have the .vscode directory included, right?

**Comment by Zaubeerer** [comment]
> Thanks, it's common to have the .vscode directory included, right?

It depends! 

I like the idea of adding a `.devcontainer`, `launch.json` and `.vscode` in general as configuration so that all members of the team can easily work with the same setup.

For sure I wanted to include it here, so people can easily set up the debug configurations for their projects :)

**Comment by Zaubeerer** [comment]
@bbelderbos, I finally fixed some confusions about the URLs used for testing and development.

This should be mergeable now, after your review.

File: frontend/pages/update.py [hunk 1]
```diff
@@ -1,7 +1,12 @@
 import requests
 import streamlit as st
+from decouple import config
 
-API_URL = "http://localhost:8000/decisions/"
+DEBUG = config("DEBUG", default=False, cast=bool)
```
Comment (bbelderbos) [inline frontend/pages/update.py:5]: could define those once in a constants.py

File: frontend/requirements.txt [hunk 2]
```diff
@@ -1,3 +1,5 @@
 streamlit>=1.15.0
 pandas>=1.3
 pytest-playwright
+pytest-order
```
Comment (bbelderbos) [inline frontend/requirements.txt:4]: best to pin them

File: frontend/utils.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import pandas as pd
+import requests
+from decouple import config
+
+DEBUG = config("DEBUG", default=False, cast=bool)
+
+API_URL = (
```
Comment (bbelderbos) [inline frontend/utils.py:7]: also repeated here

**Review by bbelderbos — APPROVED** [review]
Thanks! No need to make any changes, this is great for people wanting to follow along with the session.

**Comment by Zaubeerer** [comment]
@bbelderbos, at some point, we could simply do a "refactor decisions app" code clinic :)

**Comment by bbelderbos** [comment]
That would be nice yes, even if simple refactorings + showing the approach + using an existing code base is all valuable. Listing it in our sheet.

**Pull request merged**